scala-package/macros/src/main/scala/org/apache/mxnet/NDArrayMacro.scala

```diff
@@ -69,7 +69,7 @@ private[mxnet] object NDArrayMacro {
             // e.g def transpose(kwargs: Map[String, Any] = null)(args: Any*)
             q"def $termName(kwargs: Map[String, Any] = null)(args: Any*) = {genericNDArrayFunctionInvoke($funcName, args, kwargs)}".asInstanceOf[DefDef],
             // e.g def transpose(args: Any*)
-            q"def $termName(args: Any*) = {genericNDArrayFunctionInvoke($funcName, args, null)}".asInstanceOf[DefDef]
+            q"@scala.annotation.varargs def $termName(args: Any*) = {genericNDArrayFunctionInvoke($funcName, args, null)}".asInstanceOf[DefDef]
```

Comment:
Why do we need to all APIs to have @vaargs. This will mean we will have to maintain this functionality going forward and I don't think this is an optimal solution. -- We are trying to get rid of this sooner and move to the new APIs right?

Reply:
What's the new APIs for Java? For now the type-safe API can not directly used by java because of default args.
I think we can safely deprecate this API (for both Scala and Java) once we have a complete solution (for both Scala and Java)